### ISSUE TITLE
snowflake_database_grant causes drift each time

### ISSUE THREAD

**Issue by jglapa** [Snowflake-Labs/terraform-provider-snowflake#1611]
**Provider Version**

`0.58.0`

**Terraform Version**

`v1.3.6`

**Describe the bug**

With this resource definition (see code samples) after apply the `snowflake_database_grant.test_database_grant` is always raised as not applied! Subsequent apply don't help.

```
  # snowflake_database_grant.test_database_grant will be updated in-place
  ~ resource "snowflake_database_grant" "test_database_grant" {
        id                     = "SNOWFLAKE|IMPORTED PRIVILEGES|false|TEST_ROLE|"
      ~ roles                  = [
          + "TEST_ROLE",
        ]
        # (5 unchanged attributes hidden)
    }
```

`state show`:
```hcl
# snowflake_database_grant.test_database_grant:
resource "snowflake_database_grant" "test_database_grant" {
    database_name          = "SNOWFLAKE"
    enable_multiple_grants = false
    id                     = "SNOWFLAKE|IMPORTED PRIVILEGES|false|TEST_ROLE|"
    privilege              = "IMPORTED PRIVILEGES"
    roles                  = [
        "TEST_ROLE",
    ]
    shares                 = []
    with_grant_option      = false
}
```

**Expected behavior**

`snowflake_database_grant` is applied and not reported as diff

**Code samples and commands**

```hcl
resource "snowflake_role" "test_role" {
  name = "TEST_ROLE"
}

resource "snowflake_database_grant" "test_database_grant" {
  database_name     = "SNOWFLAKE"
  privilege         = "IMPORTED PRIVILEGES"
  roles             = [
    snowflake_role.test_role.name
  ]
}
```

**Additional context**

Started happening after upgrade to `0.58.0` from `0.49.0`. There were more issue with the grant resource I believe because of different handling of the `id` field. `state rm` and `apply` helped for those but the one reported doesn't want to go away.
Similar problem I believe here: https://github.com/Snowflake-Labs/terraform-provider-snowflake/issues/1573

**Comment by Relativity74205** [comment]
https://github.com/Snowflake-Labs/terraform-provider-snowflake/pull/1597 was merged into main and is scheduled to be released with 0.59.0.

**Comment by VickyWinner** [comment]
Is it just me?  It's granting on first terraform apply and revoking when I run terraform apply again.

**Comment by jglapa** [comment]
> Is it just me? It's granting on first terraform apply and revoking when I run terraform apply again.

@VickyWinner are you upgrading from an older version? I think the identifier of the resource was changed somewhere around version 0.55 and has now more fields. 
The remedy at least for me was to remove the resource from the state and import it. That resolved the issue

**Comment by jglapa** [comment]
issue was fixed in v0.59

**Comment by VickyWinner** [comment]
I can destroy all and try again with 0.60.0.  The `snowflake_database_grant` is for new database creating by TF.  Will try that and see.
